src/test/kotlin/org/jmailen/gradle/kotlinter/functional/KotlinProjectTest.kt

```diff
@@ -186,6 +186,30 @@ internal class KotlinProjectTest : WithGradleTest.Kotlin() {
         }
     }
 
+    @Test
+    fun `plugin resolves custom dynamically loaded RulesetProviders`() {
+        settingsFile()
+        buildFile()
+        editorConfig()
+        kotlinSourceFile(
+            "CustomObject.kt",
+            """
+            object CustomObject
+            
+            """.trimIndent(),
+        )
+
+        build("lintKotlin", "--info").apply {
+            assertEquals(SUCCESS, task(":lintKotlin")?.outcome)
+            assertTrue(output.contains("Resolved 45 RuleSetProviders"))
+        }
+
+        build("formatKotlin", "--info").apply {
+            assertEquals(SUCCESS, task(":formatKotlin")?.outcome)
+            assertTrue(output.contains("Resolved 45 RuleSetProviders"))
+        }
```

Comment:
This is not ideal tests, and will likely fail when upgrading ktlint version, but I was out of ideas and decided to defer writing better assertion to somewhen _later_ 😬 
What I want to test here is to somehow replace the old test relying on test classpath, with something that covers the dynamic loading part.